include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -231,7 +342,26 @@ class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
 			return true;
 		}
 
+		inline size_t getTotalMutableSamplerCount() const { return m_mutableSamplerCount; }
+		inline size_t getTotalDescriptorCount(const E_DESCRIPTOR_TYPE type) const { return m_descriptorCount[type]; }
+
 		core::SRange<const SBinding> getBindings() const { return {m_bindings->data(), m_bindings->data()+m_bindings->size()}; }
+
+		inline uint32_t getBindingCount(const E_DESCRIPTOR_TYPE type) const { return m_descriptorRedirects[type].count; }
+		inline uint32_t* getBindingStorage(const E_DESCRIPTOR_TYPE type) const { return m_descriptorRedirects[type].bindings.get(); }
+		inline uint32_t* getBindingOffsetStorage(const E_DESCRIPTOR_TYPE type) const { return m_descriptorRedirects[type].offsets; }
+
+		inline uint32_t getDescriptorOffset(const E_DESCRIPTOR_TYPE type, const uint32_t binding) const{ return m_descriptorRedirects[type][binding]; }
+		inline uint32_t getMutableSamplerOffset(const uint32_t binding) const { return m_samplerRedirects[binding]; }
+
+	protected:
+		// Maps a binding number to a local (to descriptor set layout) offset, for a given descriptor type.
+		SBindingRedirect m_descriptorRedirects[asset::EDT_COUNT];
+		SBindingRedirect m_samplerRedirects;
+
+	private:
+		size_t m_mutableSamplerCount = 0ull;
+		size_t m_descriptorCount[EDT_COUNT];
```

Comment:
store these inside `SBindingRedirect` then you can implement `getCount()` for a binding

Reply:
Resolved.